requirements-dev.txt

```diff
@@ -1,120 +1,51 @@
-alembic==1.4.1
-appdirs==1.4.4
-appnope==0.1.2
-argh==0.26.2
-attrs==20.3.0
-azure-core==1.9.0
-azure-storage-blob==12.6.0
-backcall==0.2.0
+argh>=0.26
+autoapi>=2.0.1
 black==20.8b1
-boto3==1.16.28
-botocore==1.19.28
-certifi==2020.11.8
-cffi==1.14.4
-cfgv==3.2.0
-chardet==3.0.4
+boto3==1.16.63
+botocore==1.19.63
+certifi==2020.12.5
 click==8.0.0a1
-cloudpickle==1.6.0
-colorama==0.4.4
-coverage==5.3
-cryptography==3.3.2
-cycler==0.10.0
-scikit-learn==0.24.1
-databricks-cli==0.14.1
-decorator==4.4.2
-distlib==0.3.1
-docker==4.4.0
-docutils==0.16
-entrypoints==0.3
-filelock==3.0.12
-Flask==1.1.2
-fsspec==0.8.4
-gitdb==4.0.5
-GitPython==3.1.11
-gunicorn==20.0.4
-identify==1.5.10
-idna==2.10
-importlib-metadata==3.1.1
-iniconfig==1.1.1
+coverage==5.4
+flake8==3.9.0
+freezegun==1.1.0
+fsspec==0.8.5
 ipykernel==5.3.4
-ipython==7.19.0
-ipython-genutils==0.2.0
-isodate==0.6.0
-itsdangerous==1.1.0
 jmespath==0.10.0
-jupyter-client==6.1.7
-jupyter-core==4.7.0
-keyring==21.5.0
-kiwisolver==1.3.1
-Mako==1.1.3
-MarkupSafe==1.1.1
-marshmallow==3.9.1
+marshmallow==3.10.0
 matplotlib==3.3.3
-mlflow==1.12.1
-msrest==0.6.19
-mypy-extensions==0.4.3
-nodeenv==1.5.0
+mlflow==1.13.1
+moto==1.3.16
 numpy==1.19.4
-oauthlib==3.1.0
-packaging==20.7
+openpyxl==3.0.6
 pandas==1.2.1
-parso==0.7.1
-pathspec==0.8.1
-pexpect==4.8.0
-pickleshare==0.7.5
 Pillow==8.1.1
-pkginfo==1.6.1
-pluggy==0.13.1
 pre-commit==2.9.2
-prometheus-client==0.9.0
-prometheus-flask-exporter==0.18.1
-prompt-toolkit==3.0.8
 protobuf==3.15.6
-ptyprocess==0.6.0
-py==1.10.0
+puremagic ==1.10
 pyarrow==2.0.0
-pycparser==2.20
-Pygments==2.7.2
-pyparsing==2.4.7
-pytest==6.1.2
-pytest-cov==2.10.1
-pytest-runner==5.3
+pytest-cov>=2.11.1
+pytest-runner>=5.2
+pytest==6.2.2
 python-dateutil==2.8.1
-python-editor==1.0.4
-pytz==2020.4
+pytz==2020.5
 PyYAML==5.4
-pyzmq==20.0.0
-querystring-parser==1.2.4
-readme-renderer==28.0
-regex==2020.11.13
-requests==2.25.0
-requests-oauthlib==1.3.0
-requests-toolbelt==0.9.1
-rfc3986==1.4.0
-s3transfer==0.3.3
+s3transfer==0.3.4
+scikit-learn==0.24.1
 setuptools-black==0.1.5
 six==1.15.0
-smmap==3.0.4
-SQLAlchemy==1.3.20
-sqlparse==0.4.1
-tabulate==0.8.7
-toml==0.10.2
-tornado==6.1
+smart-open==4.1.2
+sphinx-autoapi>=1.4.0
```

Comment:
we shouldn't have sphinx devs in our dev dependencies. This is for documentation dependencies and they are separate

Reply:
i disagree, my understanding is that this was included as a full dependency for a developer working on this project. which what the team has been asking for. otherwise lets just remove this file.

Reply:
but agree it shouldn't be part of the main requirements